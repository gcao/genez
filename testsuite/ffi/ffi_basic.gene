# Test basic FFI declarations and calls

# Declare printf from libc
(c-extern printf ([format "char*"]) "int" "libc" :variadic true)

# Test printf call
(printf "Hello from FFI!")
(print "")  # Add newline using print

# Declare sin from libm
(c-extern sin ([x "double"]) "double" "libm")

# Test sin call
(var angle 1.5708)
(var result (sin angle))
(print "sin(" angle ") = " result)

# Test c-struct declaration
(c-struct Point ([x "double"] [y "double"]))

# Test c-type declaration
(c-type size_t "unsigned long")