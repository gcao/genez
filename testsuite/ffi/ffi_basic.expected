Hello from FFI!
sin( 1.5708 ) =  0.9999999999932537